modules/extension/xsd/xsd-wfs/src/main/java/org/geotools/wfs/PropertyValueCollection.java

```diff
@@ -180,7 +177,8 @@ public Object next() {
                             descriptor.isNillable(),
                             descriptor.getDefaultValue());
 
-            if (next instanceof SimpleFeature) {
+            if (next instanceof SimpleFeature || !(value instanceof ComplexAttribute)) {
```

Comment:
Humm, something looks strange here. [``Attribute``](https://github.com/geotools/geotools/blob/41ac131a56b020bd6bf1436e8fd0fa303f7927d1/modules/library/opengis/src/main/java/org/opengis/feature/Attribute.java#L51) is the parent class of [``ComplexAttribute``](https://github.com/geotools/geotools/blob/41ac131a56b020bd6bf1436e8fd0fa303f7927d1/modules/library/opengis/src/main/java/org/opengis/feature/ComplexAttribute.java#L90), hence its more generic.

What we are expressing here is ignore this specific case, but for all the others including this one, perform a  ``((Attribute) value).getValue()``. I think we should test the value class rather than the feature class.